app/code/Magento/CustomerImportExport/Model/Import/Customer.php

```diff
@@ -569,11 +569,11 @@ public function getEntityTable()
      */
     public function getValidColumnNames()
     {
-        $this->validColumnNames = array_merge(
-            $this->validColumnNames,
-            $this->customerFields
+        return array_unique(
+            array_merge(
+                $this->validColumnNames,
+                $this->customerFields
```

Comment:
I'm not saying calling `array_flip`, I'm saying to flip arrays by themselves.

So, instead of `['column-name']` there will be `['column-name' => 1]`.

> Where is the complexity here?

I'm saying about https://en.wikipedia.org/wiki/Time_complexity#Table_of_common_time_complexities